src/renderer/redux/sagas/lcp.ts

```diff
@@ -7,14 +7,31 @@
 
 import { lcpActions } from "readium-desktop/common/redux/actions";
 import { open } from "readium-desktop/common/redux/actions/dialog";
+import { diRendererGet } from "readium-desktop/renderer/di";
 import { SagaIterator } from "redux-saga";
 import { all, call, put, take } from "redux-saga/effects";
 
+// import { PublicationView } from "readium-desktop/common/views/publication";
+// import { ReaderState } from "readium-desktop/renderer/redux/states/reader";
+
 export function* lcpUserKeyCheckRequestWatcher(): SagaIterator {
     while (true) {
         const action: any = yield take(lcpActions.ActionType.UserKeyCheckRequest);
 
         const { hint, publication, message } = action.payload;
+        // const id1 = (publication as PublicationView).identifier;
+
+        const store = diRendererGet("store");
```

Comment:
Have you thought about that : https://redux-saga.js.org/docs/api/index.html#selectselector-args

Reply:
I would love to clean this up. Right now testing for a truthy condition of `rendererStore.getState().reader.reader` is totally hacky!
In fact, the "LCP passphrase" and "publication information" (etc.) dialogs do not trigger using the same technique, there are inconsistencies in the current implementation which make it hard to track the action flow (as a developer analysing the code).
Thanks for you tips, keep them coming! :)